api/v2/Routes/students/masterymapping/index.js

```diff
@@ -0,0 +1,75 @@
+import { Router } from 'express';
+import {
+    getMaxScores,
+    getStudentScores,
+} from '../../../../lib/redisHelper.mjs';
+import ProgressReportData from '../../../../assets/progressReport/CS10.json' assert { type: 'json' };
+
+const router = Router({ mergeParams: true });
+
+function getTopicsFromUser(gradeData) {
+    const topicsTable = {};
+    Object.entries(gradeData).forEach(([assignment, topics]) => {
+        Object.entries(topics).forEach(([topic, score]) => {
+            if (topic in topicsTable) {
+                topicsTable[topic] += +(score ?? 0);
+            } else {
+                topicsTable[topic] = +(score ?? 0);
+            }
+        });
+    });
```

Comment:
Would be nice to do this in $O(N)$ using an accumulator function instead of $O(N^2)$.

Reply:
Modified!